src/org/java/training/StackDemo.java

```diff
@@ -0,0 +1,113 @@
+package org.java.training;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Self-Test question 12.
+ */
+public class StackDemo {
+	public static void main(String... args) {
+		Stack<String> stack = new FixedStack<>(10, "Flame", "Bubblegum", "Lumpy Space", "Ghost", "Slime", "Breakfast", "Crab", "Wildberry");
+		stack.push("Turtle").push("Hot Dog");
+		
+		stack.stream().forEach(System.out::println);
+
+		try {
+			stack.push("Bee");
+		} catch( StackOverflowException e ) {
+			e.printStackTrace();
+		}
+		
+		System.out.println();
+		
+		while(stack.hasNext())
+			System.out.println(stack.pop());
+		
+		stack.pop();
+	}
+}
+
+interface Stack<T> extends Iterable<T> {
+	Stack<T> push(T element);
+	T pop();
+	boolean hasNext();
+
+	default Stream<T> stream() {
+		return StreamSupport.stream(spliterator(), false);
+	}
+}
+
+class StackOverflowException extends RuntimeException {
+	private static final long serialVersionUID = 1L;
+
+	public StackOverflowException(int size) {
+		super( String.format("Insertion attempt exceeds maximum stack size [%d]", size) );
```

Comment:
you should have had the error message say: `"This post does not meet our quality standards."`